pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -333,21 +346,42 @@ func (c *Controller) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthE
 		return
 	}
 
-	// get previous status
-	cfg := c.store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
-	if cfg == nil {
-		log.Errorf("config was nil when getting WorkloadEntry %v for %v", entryName, proxy.ID)
-		return
-	}
+	condition := transformHealthEvent(event)
+	c.storeHealthStatus(makeProxyKey(proxy), condition)
 
-	// replace the updated status
-	wle := status.UpdateConfigCondition(*cfg, transformHealthEvent(event))
-	// update the status
-	_, err := c.store.UpdateStatus(wle)
-	if err != nil {
-		log.Errorf("error while updating WorkloadEntry status: %v for %v", err, proxy.ID)
+	// backoff retry
+	for i := 0; i < maxRetries; i++ {
+		if i > 0 {
+			time.Sleep(time.Duration(float64(waitTime.Nanoseconds()) * math.Pow(2, float64(i-1))))
```

Comment:
we use github.com/cenkalti/backoff.NewExponentialBackOff() in other places if you prefer

Reply:
good to know